length_field.go

```diff
@@ -31,22 +31,43 @@ func (l *lengthField) check(curOffset int, buf []byte) error {
 type varintLengthField struct {
 	startOffset int
 	length      int64
+	adjusted    bool
+	size        int
```

Comment:
size could always be calculated, but it involves encoding the varint into a buffer, so not very cheap.

Reply:
It looks pretty cheap to me? The buffer doesn't escape so it will be allocated on the stack, and the actual encoding itself is just a couple of bit-ops in a very short for-loop.